File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    scores = {"R": 0, "T": 0, "C": 0, "F": 0, "J": 0, "M": 0, "A": 0, "N": 0}
    
    for s, c in zip(survey, choices):
        if c < 4:
            scores[s[0]] += (4 - c)
        elif c > 4:
            scores[s[1]] += (c - 4)
            
    answer = ""
    for type1, type2 in [('R', 'T'), ('C', 'F'), ('J', 'M'), ('A', 'N')]:
        if scores[type1] >= scores[type2]:
            answer += type1
        else:
            answer += type2
            
    return answer

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

char* solution(const char* survey[], size_t survey_len, int choices[], size_t choices_len) {
    int scores[26] = {0};
    
    for (size_t i = 0; i < survey_len; i++) {
        int choice = choices[i];
        
        if (choice < 4) {
            scores[survey[i][0] - 'A'] += (4 - choice);
        } else if (choice > 4) {
            scores[survey[i][1] - 'A'] += (choice - 4);
        }
    }
    
    char* answer = (char*)malloc(5 * sizeof(char));

    answer[0] = (scores['R' - 'A'] >= scores['T' - 'A']) ? 'R' : 'T';
    answer[1] = (scores['C' - 'A'] >= scores['F' - 'A']) ? 'C' : 'F';
    answer[2] = (scores['J' - 'A'] >= scores['M' - 'A']) ? 'J' : 'M';
    answer[3] = (scores['A' - 'A'] >= scores['N' - 'A']) ? 'A' : 'N';
    answer[4] = '\0';
    
    return answer;
}